controllers/machine_controller.go

```diff
@@ -339,32 +350,26 @@ func (r *MachineReconciler) drainNode(cluster *clusterv1.Cluster, nodeName strin
 
 func (r *MachineReconciler) deleteNode(ctx context.Context, cluster *clusterv1.Cluster, name string) error {
 	logger := r.Log.WithValues("machine", name)
-
+	var c client.Client
+	var err error
 	if cluster == nil {
 		// Try to retrieve the Node from the local cluster, if no Cluster reference is found.
-		var node corev1.Node
-		if err := r.Client.Get(ctx, client.ObjectKey{Name: name}, &node); err != nil {
-			return err
+		c = r.Client
+	} else {
+		logger = logger.WithValues("cluster", cluster.Name, "namespace", cluster.Namespace)
+		// Otherwise, proceed to get the remote cluster client to get the Node.
+		c, err = remote.NewClusterClient(r.Client, cluster, r.scheme)
+		if err != nil {
+			logger.Error(err, "Error creating a remote client for cluster while deleting Machine, won't retry")
+			return nil
 		}
-		return r.Client.Delete(ctx, &node)
 	}
 
-	logger = logger.WithValues("cluster", cluster.Name, "namespace", cluster.Namespace)
-
-	// Otherwise, proceed to get the remote cluster client and get the Node.
-	remoteClient, err := remote.NewClusterClient(r.Client, cluster)
-	if err != nil {
-		logger.Error(err, "Error creating a remote client for cluster while deleting Machine, won't retry")
-		return nil
-	}
-
-	corev1Remote, err := remoteClient.CoreV1()
-	if err != nil {
-		logger.Error(err, "Error creating a remote client for cluster while deleting Machine, won't retry")
-		return nil
+	var node corev1.Node
```

Comment:
```suggestion
	node := &corev1.Node{}
```

To be consistent with the code below